### PR TITLE
fix: remove pending ingress log statement for tcp port

### PR DESCRIPTION
**Pull request by jsilverio22**
Issue: #715 
Signed-off-by: Joshua Silverio <joshua@acorn.io>

### CODE REVIEW

File: pkg/controller/appdefinition/endpoints.go [hunk 1]
```diff
@@ -68,7 +68,7 @@ func serviceEndpoints(req router.Request, app *v1.AppInstance) (endpoints []v1.E
 				endpoints = append(endpoints, v1.Endpoint{
 					Target:     containerName,
 					TargetPort: port.TargetPort.IntVal,
-					Address:    fmt.Sprintf("<Pending Ingress>:%d", port.Port),
+					Address:    fmt.Sprintf("<Pending>:%d", port.Port),
```
Comment (cjellick) [inline pkg/controller/appdefinition/endpoints.go:71]: This is actually slightly more complicated...there are two ways that end points are exposed:
1. If the port being published is an HTTP port, ingress is used
2. If the port being published is something else, like a TCP port, a service load balancer is used

I'd like this message to reflect that. I think the port should have the protocol in there somewhere.

**Review by cjellick — DISMISSED** [review]
See inline comment

File: pkg/controller/appdefinition/endpoints.go (hunk 1 above)
Comment (jsilverio22) [reply pkg/controller/appdefinition/endpoints.go]:
```
jsilverio@joshuas-mbp acorn % ./bin/acorn run -p 9000 tcp
cool-violet
STATUS: ENDPOINTS[] HEALTHY[] UPTODATE[] 
STATUS: ENDPOINTS[] HEALTHY[0] UPTODATE[0] pending
STATUS: ENDPOINTS[] HEALTHY[0/1] UPTODATE[1] [containers: echo ContainerCreating]
STATUS: ENDPOINTS[tcp://<Pending> => echo:9000] HEALTHY[0/1] UPTODATE[1] [containers: echo ContainerCreating]
STATUS: ENDPOINTS[tcp://127.0.0.1:9000 => echo:9000] HEALTHY[0/1] UPTODATE[1] [containers: echo ContainerCreating]

┌────────────────────────────────────────────────────────────────────────────────┐
| STATUS: ENDPOINTS[tcp://127.0.0.1:9000 => echo:9000] HEALTHY[1] UPTODATE[1] OK |
└────────────────────────────────────────────────────────────────────────────────┘
jsilverio@joshuas-mbp acorn % ./bin/acorn                
jsilverio@joshuas-mbp acorn % acorn run -P ghcr.io/acorn-io/library/hello-world

cold-fog
STATUS: ENDPOINTS[] HEALTHY[] UPTODATE[] 
STATUS: ENDPOINTS[] HEALTHY[0] UPTODATE[0] pending
STATUS: ENDPOINTS[] HEALTHY[0/1] UPTODATE[1] [containers: webapp ContainerCreating]
STATUS: ENDPOINTS[http://<Pending Ingress> => webapp:80] HEALTHY[0/1] UPTODATE[1] [containers: webapp ContainerCreating]
STATUS: ENDPOINTS[http://<Pending Ingress> => webapp:80] HEALTHY[0/1] UPTODATE[1] [containers: webapp is not ready]
┌────────────────────────────────────────────────────────────────────────────────────┐
| STATUS: ENDPOINTS[http://<Pending Ingress> => webapp:80] HEALTHY[1] UPTODATE[1] OK |
```

Is this more the functionality you were expecting? @cjellick

File: pkg/controller/appdefinition/status.go [hunk 2]
```diff
@@ -405,7 +405,12 @@ func endpoints(req router.Request, cfg *apiv1.Config, app *v1.AppInstance) (stri
 			}
 
 			if endpoint.Pending {
-				buf.WriteString("<Pending Ingress>")
+				switch endpoint.Protocol {
+				case "tcp":
```
Comment (ibuildthecloud) [inline pkg/controller/appdefinition/status.go:409]: Technically I believe "UDP" is also supported.  I would probably just check for http and set "Pending Ingress" and for everything else "Pending Load Balancer"

File: pkg/controller/appdefinition/endpoints.go [hunk 3]
```diff
@@ -72,6 +72,7 @@ func serviceEndpoints(req router.Request, app *v1.AppInstance) (endpoints []v1.E
 					Protocol:   protocol,
 					Pending:    true,
 				})
+
```
Comment (ibuildthecloud) [inline pkg/controller/appdefinition/endpoints.go:75]: This can be removed

**Review by tylerslaton — APPROVED** [review]
Took a look locally and the change looks runs as expected. 👍🏻

**Review by ibuildthecloud — APPROVED** [review]

**Pull request merged**